src/14_bc_connectivity_data.R

```diff
@@ -0,0 +1,668 @@
+#
+#
+# References
+#
+# National Broadband Data (NBD) overview and methodology, including pseudo‑households and precision at 250m: Innovation, Science and Economic Development Canada. [ised-isde.canada.ca]
+# 2021 Census Boundary Files (CSD polygons, NAD83 LCC) and relationship file: Statistics Canada, Open Government. [open.canada.ca]
+
+# what those terms mean in the context of **National Broadband Data (NBD)** and Canada’s broadband programs:
+#
+#   ### **Government Support**
+#
+#   Refers to broadband projects funded or supported by **government programs** (federal, provincial, or territorial) to expand high-speed Internet in underserved areas. These include initiatives like the **Universal Broadband Fund (UBF)** and CRTC’s **Broadband Fund**, which provide grants to build infrastructure in rural and remote communities. [\[crtc.gc.ca\]](https://crtc.gc.ca/eng/internet/fnds.htm), [\[ised-isde.canada.ca\]](https://ised-isde.canada.ca/site/high-speed-internet-canada/en/universal-broadband-fund)
+#
+# ***
+#
+#   ### **Private Expansions**
+#
+#   Indicates projects led by **private Internet Service Providers (ISPs)** without direct government funding. These expansions are market-driven, where ISPs invest their own capital to extend broadband coverage into new areas, often where demand or business viability exists. [\[ised-isde.canada.ca\]](https://ised-isde.canada.ca/site/high-speed-internet-canada/en/universal-broadband-fund/national-broadband-data-information)
+#
+# ***
+#
+#   ### **UBF Core**
+#
+#   Stands for **Universal Broadband Fund – Core Stream**, the main funding program launched by the Government of Canada in 2020 with a $3.225 billion budget. Its goal is to provide **50/10 Mbps** service to 98% of Canadian households by 2026 and 100% by 2030. The Core stream supports large-scale infrastructure projects in rural and remote communities. [\[cnpartners.ca\]](https://cnpartners.ca/2025-report-the-state-of-canadas-universal-broadband-fund/), [\[ised-isde.canada.ca\]](https://ised-isde.canada.ca/site/high-speed-internet-canada/en/universal-broadband-fund)
+#
+# ***
+#
+#   ### **UBF RRS**
+#
+#   Refers to the **Universal Broadband Fund – Rapid Response Stream**, a fast-track funding stream for **shovel-ready projects** that can be deployed quickly in areas with urgent connectivity needs. It prioritizes projects that can deliver immediate improvements, often in rural or Indigenous communities. [\[ised-isde.canada.ca\]](https://ised-isde.canada.ca/site/high-speed-internet-canada/en/universal-broadband-fund), [\[www.advintive.com\]](http://www.advintive.com/ubf-rrs/)
+#
+# ***
+#
+# ------------------
+
+# Packages
+library(forcats)
+# --- Aggregate NBD-PHH-Speeds to Census Subdivision (CSD) level ---
+# Packages
+library(sf)
+library(dplyr)
+library(readr)
+library(stringr)
+library(tidyr)
+library(lubridate)
+library(glue)
+library(datadictionary)
+source("./src/utils.R") # get the functions for plotting maps
```

Comment:
need to source ggplot2 either in this utils script or above - errors out otherwise.